sdks/java/io/hdfs/src/main/java/org/apache/beam/sdk/io/hdfs/HDFSFileSource.java

```diff
@@ -75,167 +91,174 @@
+    SerializableSplit serializableSplit = new SerializableSplit(inputSplit);
+    return new HDFSFileSource<>(
+        filepattern, formatClass, coder, inputConverter,
+        serializableConfiguration, serializableSplit, validate);
   }
 
-  protected List<InputSplit> computeSplits(long desiredBundleSizeBytes) throws IOException,
-      IllegalAccessException, InstantiationException {
-    Job job = Job.getInstance();
-    FileInputFormat.setMinInputSplitSize(job, desiredBundleSizeBytes);
-    FileInputFormat.setMaxInputSplitSize(job, desiredBundleSizeBytes);
-    return createFormat(job).getSplits(job);
+  public HDFSFileSource<T, K, V> withoutValidation() {
+    return new HDFSFileSource<>(
+        filepattern, formatClass, coder, inputConverter,
+        serializableConfiguration, serializableSplit, false);
   }
 
-  @Override
-  public BoundedReader<KV<K, V>> createReader(PipelineOptions options) throws IOException {
-    this.validate();
+  // =======================================================================
+  // BoundedSource
+  // =======================================================================
 
+  @Override
+  public List<? extends BoundedSource<T>> splitIntoBundles(
+      long desiredBundleSizeBytes,
+      PipelineOptions options) throws Exception {
     if (serializableSplit == null) {
-      return new HDFSFileReader<>(this, filepattern, formatClass);
+      return Lists.transform(computeSplits(desiredBundleSizeBytes),
+          new Function<InputSplit, BoundedSource<T>>() {
+            @Override
+            public BoundedSource<T> apply(@Nullable InputSplit inputSplit) {
+              SerializableSplit serializableSplit = new SerializableSplit(inputSplit);
+              return new HDFSFileSource<>(
+                  filepattern, formatClass, coder, inputConverter,
+                  serializableConfiguration, serializableSplit, validate);
+            }
+          });
     } else {
-      return new HDFSFileReader<>(this, filepattern, formatClass,
-          serializableSplit.getSplit());
-    }
-  }
-
-  @Override
-  public Coder<KV<K, V>> getDefaultOutputCoder() {
-    return KvCoder.of(getDefaultCoder(keyClass), getDefaultCoder(valueClass));
-  }
-
-  @SuppressWarnings("unchecked")
-  private <T> Coder<T> getDefaultCoder(Class<T> c) {
-    if (Writable.class.isAssignableFrom(c)) {
-      Class<? extends Writable> writableClass = (Class<? extends Writable>) c;
-      return (Coder<T>) WritableCoder.of(writableClass);
-    } else if (Void.class.equals(c)) {
-      return (Coder<T>) VoidCoder.of();
+      return ImmutableList.of(this);
     }
-    // TODO: how to use registered coders here?
-    throw new IllegalStateException("Cannot find coder for " + c);
   }
 
-  // BoundedSource
-
   @Override
-  public long getEstimatedSizeBytes(PipelineOptions options) {
+  public long getEstimatedSizeBytes(PipelineOptions options) throws Exception {
```

Comment:
In this case the estimation is best effort, and this `throws Exception` looks like a leftover from before the refactor. Will remove.